### PR TITLE
pyright updates for model.py and entity.py

### PR DESCRIPTION
**Pull request by jwhonce**
* Flagged internal items with # type: ignore
* Flagged optional exported properties
* Update typing imports

### CODE REVIEW

**Comment by mm21** [comment]
Hi, thanks for submitting this! I've been doing some typing work myself lately so went ahead and merged that.

I think most of these `# type: ignore`s should actually be properly fixed. IMO that should generally only be used for special cases like descriptors, wherein there's a fundamental impasse between the API and implementation.

There are a couple different classes of common problems, and I see you've already addressed some of them this way, but I think this approach can be applied to the remaining `# type: ignore`s as well.

1. Has a lifecycle where it may be `None`: add `| None` to the type

Examples:
  - Before newly created entities are flushed, their `_entity_id` is `None`
```python
    _entity_id: str = None
```

- Model states will be `None` until setup (e.g. upon being accessed by user), then can become `None` again later (e.g. upon `invalidate()`)
```python
    _backing: dict[str, str | int | bool] = None
```

2. Shouldn't actually be assigned as `None`: remove ` = None`
Essentially if it's initialized in `__init__`, then it can be simply declared - it's not expected to be `None` at any point during its lifecycle.

Example:
  - Since `_state` is initialized in `__init__`, it should have a valid `State` during its whole lifecycle
```python
    _state: State = None
```

Some of them may be more tricky than others, or require more in-depth knowledge of the design. Feel free to leave those and I'll be happy to work on them.

File: trilium_alchemy/core/entity/model.py [hunk 1]
```diff
@@ -78,31 +76,31 @@ class Model(ABC):
     """
 
     # pydantic model used in etapi
-    etapi_model: Type[BaseModel] = None
+    etapi_model: Type[BaseModel] | None = None
```
Comment (mm21) [inline trilium_alchemy/core/entity/model.py:79]: These ones (through `fields_default`) should be overridden by subclasses, so should never be `None` - we can just remove the assignment

File: trilium_alchemy/core/entity/model.py [hunk 2]
```diff
@@ -171,18 +172,24 @@ def fields_update_alias(cls) -> Iterable[str]:
         Maps aliased fields to canonical fields using fields_alias if
         provided.
         """
+
+        if cls.fields_update is None:
```
Comment (mm21) [inline trilium_alchemy/core/entity/model.py:176]: Prefer `assert cls.fields_update is not None` to avoid further unexpected behavior if this hits

**Comment by jwhonce** [comment]
With changes in trilium-notes future, I have moved to logseq.

**Pull request closed without merging**